pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -66,17 +66,6 @@ func newECRProvider(getterFactory tokenGetterFactory) *ecrProvider {
 // Enabled implements DockerConfigProvider.Enabled. Enabled is true if AWS
 // credentials are found.
 func (p *ecrProvider) Enabled() bool {
-	sess, err := session.NewSessionWithOptions(session.Options{
-		SharedConfigState: session.SharedConfigEnable,
-	})
-	if err != nil {
-		klog.Errorf("while validating AWS credentials %v", err)
-		return false
-	}
-	if _, err := sess.Config.Credentials.Get(); err != nil {
-		klog.Errorf("while getting AWS credentials %v", err)
-		return false
-	}
 	return true
```

Comment:
this means that Provide() will be called for all images for all providers and log v(3) output about the image not being a valid ECR repository URL for non-ECR images

Would it make sense to also:

1. drop the `klog.V(3).Info(err)` output if a non-ECR image is passed to provide... this isn't something we need to log

2. move the current Enabled() check to a once.Do check we call before calling `getFromECR`. If that fails, make future calls to Provider() no-op

Reply:
I think both of those sound fine to me.

Reply:
The only issue with this is we have to be careful with failures, i.e. we don't want an outage to cause permanent unhealthiness in kubelet.

Reply:
sure... by moving the existing check to a single run-once that happens only after we are asked to provide credentials for an ECR image, I *think* we do the following:
* maintain current behavior on AWS
* remove lag on non-AWS on a cluster that never tries to pull an ECR image
* limits lag on a non-AWS cluster that tries to pull an ECR image to a single attempt